lib/ex_webrtc/peer_connection.ex

```diff
@@ -316,20 +321,61 @@ defmodule ExWebRTC.PeerConnection do
   @impl true
   def handle_call({:add_transceiver, kind, options}, _from, state)
       when kind in [:audio, :video] do
-    transceiver = new_transceiver(kind, nil, options, state.config)
+    transceiver = RTPTransceiver.new(kind, nil, state.config, options)
     transceivers = state.transceivers ++ [transceiver]
 
     {:reply, {:ok, transceiver}, %{state | transceivers: transceivers}}
   end
 
   @impl true
   def handle_call({:add_transceiver, %MediaStreamTrack{} = track, options}, _from, state) do
-    transceiver = new_transceiver(track.kind, track, options, state.config)
+    transceiver = RTPTransceiver.new(track.kind, track, state.config, options)
     transceivers = state.transceivers ++ [transceiver]
 
     {:reply, {:ok, transceiver}, %{state | transceivers: transceivers}}
   end
 
+  @impl true
+  def handle_call({:add_track, %MediaStreamTrack{kind: kind} = track}, _from, state) do
+    # we ignore the condition that sender has never been used to send
+    free_transceiver =
+      Enum.find_index(state.transceivers, fn
+        %RTPTransceiver{
+          kind: ^kind,
+          sender: %RTPSender{track: nil},
+          current_direction: cr
```

Comment:
I only use short variable names for guards, otherwise, guards get horribly long and usually these variables aren used anywhere else, and if they are, more verbose names make sense. Here that's not really an issue, as the guards are split into multiple lines, but overall, I'm fine with either option.